include/tdi/common/c_frontend/tdi_table.h

```diff
@@ -0,0 +1,2364 @@
+/*******************************************************************************
+ * BAREFOOT NETWORKS CONFIDENTIAL & PROPRIETARY
+ *
+ * Copyright (c) 2017-2021 Barefoot Networks, Inc.
+
+ * All Rights Reserved.
+ *
+ * NOTICE: All information contained herein is, and remains the property of
+ * Barefoot Networks, Inc. and its suppliers, if any. The intellectual and
+ * technical concepts contained herein are proprietary to Barefoot Networks,
+ * Inc.
+ * and its suppliers and may be covered by U.S. and Foreign Patents, patents in
+ * process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material is
+ * strictly forbidden unless prior written permission is obtained from
+ * Barefoot Networks, Inc.
+ *
+ * No warranty, explicit or implicit is provided, unless granted under a
+ * written agreement with Barefoot Networks, Inc.
+ *
+ * $Id: $
+ *
+ ******************************************************************************/
+/** @file tdi_table.h
+ *
+ *  @brief Contains TDI Table APIs
+ */
+#ifndef _TDI_TABLE_H
+#define _TDI_TABLE_H
+
+#include <tdi/common/c_frontend/tdi_common.h>
+#include <tdi/common/tdi_defs.h>
+#include <tdi/common/c_frontend/tdi_table_data.h>
+#include <tdi/common/c_frontend/tdi_table_key.h>
+#include <tdi/common/c_frontend/tdi_table_operations.h>
+#include <tdi/common/c_frontend/tdi_table_attributes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Annotation struct. Contains name-value pair.
+ * This is extracted from bf-rt.json. @c value might contain
+ * null for some annotations but name will always contain
+ * a string.
+ */
+typedef struct tdi_annotation_ {
+  /** Name field of annotation */
+  const char *name;
+  /** Value field of annotation */
+  const char *value;
+} tdi_annotation_t;
+
+/**
+ * @brief Entry read flag. (Deprecated. Please use generic flag support)
+ * @deprecated Please use function version with new flags argument.
+ */
```

Comment:
Just generic question: why implemented already deprecated code in a new library?  Can't we expect contributers to use the flags arguments when contributing new code? Just asking,  as it would keep technical debt out of the code from the beginning...